### PR TITLE
Add chain_isAssetSpent RPC API

### PR DESCRIPTION
**Pull request by joojis**
SDK patch: https://github.com/CodeChain-io/codechain-sdk-js/pull/99

### CODE REVIEW

File: core/src/client/client.rs [hunk 1]
```diff
@@ -277,6 +277,42 @@ impl AssetClient for Client {
...
+
+                if index >= output_length {
+                    return Ok(None)
+                }
+
+                match Client::state_at(&self, id) {
+                    Some(state) => {
+                        let shard_id = 0;
+                        let address = AssetAddress::new(transaction_hash, index, shard_id);
+                        match state.asset(shard_id, &address)? {
```
Comment (sgkim126) [inline core/src/client/client.rs]:
```rust
Ok(Some(state.asset(shard_id, &address)?.is_none()))
```

Comment (joojis) [reply core/src/client/client.rs]: That sounds better. I will update.

**Comment by joojis** [comment]
I'm going to change this patch not to hardcode a shard id. Please be patient.

**Comment by joojis** [comment]
I updated the patch, and, I updated the SDK patch also. 
https://github.com/CodeChain-io/codechain-sdk-js/pull/99

File: core/src/client/client.rs [hunk 2]
```diff
@@ -277,6 +277,52 @@ impl AssetClient for Client {
...
         }
     }
+
+    fn is_asset_spent(
+        &self,
+        transaction_hash: H256,
+        index: usize,
+        shard_id: ShardId,
+        block_id: BlockId,
+    ) -> TrieResult<Option<bool>> {
```
Comment (sgkim126) [inline core/src/client/client.rs]: Would you add a comment about the meaning of output?

Comment (joojis) [reply core/src/client/client.rs]: I added the comment explaining the case of None.

**Pull request merged**